### PR TITLE
[moveonly] Extract CWallet::MarkInputsDirty, and privatize AddToWalletIfInvolvingMe

### PR DESCRIPTION
**Pull request by Empact**
Thus reducing code and surface area of CWallet.

### CODE REVIEW

File: src/wallet/wallet.h [hunk 1]
```diff
@@ -703,9 +703,27 @@ class CWallet final : public CCryptoKeyStore, public CValidationInterface
     void AddToSpends(const COutPoint& outpoint, const uint256& wtxid);
     void AddToSpends(const uint256& wtxid);
 
+    /**
+     * Add a transaction to the wallet, or update it.  pIndex and posInBlock should
```
Comment (promag) [inline src/wallet/wallet.h]: `... if belongs to the wallet`?

**Review by promag — COMMENTED** [review]
utACK de381c6.

File: src/wallet/wallet.h (hunk 1 above)
Comment (Empact) [reply src/wallet/wallet.h]: Going to leave as moveonly.

**Comment by Empact** [comment]
Rebased for #12944

**Comment by promag** [comment]
(re)utACK 17e6aa8.

**Comment by sipa** [comment]
utACK 17e6aa8e333dbf3631fb09dc6eb7ede1937269df

**Comment by MarcoFalke** [comment]
utACK 17e6aa8

**Pull request merged**